front_end/object_ui/CustomPreviewComponent.js

```diff
@@ -213,6 +213,37 @@ ObjectUI.CustomPreviewComponent = class {
     this.element = createElementWithClass('span', 'source-code');
     const shadowRoot = UI.createShadowRootWithCoreStyles(this.element, 'object_ui/customPreviewComponent.css');
     this.element.addEventListener('contextmenu', this._contextMenuEventFired.bind(this), false);
+
+    // DDT-specific operation to linkify an object's displayed runtime type
+    // Performed on all remote objects for which the following is resolvable:
+    // (Javascript) remoteObj["runtimeType"][Symbol(_type)]["[[FunctionLocation]]"]
+    // In Dart, this corresponds to an object's constructor's function location
+    let remoteFunction = "function remoteFunction() { " +
+        "return dart_library.debuggerLibraries()[0][\"dart\"].getReifiedType(this); }";
+    this._object._runtimeAgent.invoke_callFunctionOn({
+      objectId: this._object._objectId,
+      functionDeclaration: remoteFunction,
+      arguments: [],
+      silent: true,
+      returnByValue: false
+    }).then((remoteFunctionResult) => {
+      let remoteObject = this._object.runtimeModel().createRemoteObject(remoteFunctionResult.result);
+      remoteObject.debuggerModel().functionDetailsPromise(remoteObject).then((response) => {
```

Comment:
OK, so we invoke that remote function, its result is also a function, and then we convert that to a remote function and get its location. That should be commented.

Reply:
The pipeline for this part is RemoteFunction -> JS Object -> Remote Object. The JS Object returned by the remote function call is just wrapped in a remote object (because we can only get the function details of a remote object).